dubbo-config/dubbo-config-api/src/main/java/org/apache/dubbo/config/deploy/DefaultApplicationDeployer.java

```diff
@@ -230,9 +230,8 @@ private void registerShutdownHook() {
     private void initModuleDeployers() {
         // make sure created default module
         applicationModel.getDefaultModule();
-        // copy modules and initialize avoid ConcurrentModificationException if add new module
-        List<ModuleModel> moduleModels = new ArrayList<>(applicationModel.getModuleModels());
-        for (ModuleModel moduleModel : moduleModels) {
+        // deployer initialize
+        for (ModuleModel moduleModel : applicationModel.getModuleModels()) {
```

Comment:
Also for here.

Reply:
```java
    public List<ModuleModel> getModuleModels() {
        return Collections.unmodifiableList(moduleModels);
    }
```
getModuleModels method is return Collections.unmodifiableList, so I think that we can remove new ArrayList<>()